code/modules/power/supermatter/supermatter.dm

```diff
@@ -103,101 +103,118 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
+	var/static/gl_uid = 1
+	///Tracks the bolt color we are using
+	var/zap_icon = DEFAULT_ZAP_ICON_STATE
+	///The portion of the gasmix we're on that we should remove
 	var/gasefficency = 0.15
-
+	///Used for changing icon states for diff base sprites
 	var/base_icon_state = "darkmatter"
 
+	///Are we exploding?
 	var/final_countdown = FALSE
 
+	///The amount of damage we have currently
 	var/damage = 0
+	///The damage we had before this cycle. Used to limit the damage we can take each cycle, and for safe_alert
 	var/damage_archived = 0
+	///Our "Shit is no longer fucked" message. We send it when damage is less then damage_archived
 	var/safe_alert = "Crystalline hyperstructure returning to safe operating parameters."
+	///The point at which we should start sending messeges about the damage to the engi channels.
 	var/warning_point = 50
+	///The alert we send when we've reached warning_point
 	var/warning_alert = "Danger! Crystal hyperstructure integrity faltering!"
-	var/damage_penalty_point = 550
+	///The point at which we start sending messages to the common channel
 	var/emergency_point = 700
+	///The alert we send when we've reached emergency_point
 	var/emergency_alert = "CRYSTAL DELAMINATION IMMINENT."
+	///The point at which we delam
 	var/explosion_point = 900
+	///When we pass this amount of damage we start shooting bolts
+	var/damage_penalty_point = 550
 
-	var/emergency_issued = FALSE
-
+	///A scaling value that affects the severity of explosions.
 	var/explosion_power = 35
-	var/temp_factor = 30
-
-	var/lastwarning = 0                // Time in 1/10th of seconds since the last sent warning
+	///Time in 1/10th of seconds since the last sent warning
+	var/lastwarning = 0
+	///Refered to as eer on the moniter. This value effects gas output, heat, damage, and radiation.
 	var/power = 0
 	///Determines the rate of positve change in gas comp values
 	var/gas_change_rate = 0.05
-	///Ranging from 0 to 1, this is used to judge gas composition. It doesn't perfectly match the air around the sm, instead moving up at a rate determined by gas_change_rate per call
-	var/n2comp = 0                    // raw composition of each gas in the chamber, ranges from 0 to 1
-	var/plasmacomp = 0
-	var/o2comp = 0
-	var/co2comp = 0
-	var/n2ocomp = 0
-	var/tritiumcomp = 0
-	var/bzcomp = 0
-	var/pluoxiumcomp = 0
-	var/h2ocomp = 0
-	var/freoncomp = 0
-	var/h2comp = 0
-
-	var/pluoxiumbonus = 0
-	///used when calculating if the heat resitance from h2o is applied
-	var/h2obonus = 0
-	///it allows us to remove the power generation while freon is inside the chamber
-	var/freonbonus = 1
-
+	///The list of gases we will be interacting with in process_atoms()
+	var/list/gases_we_care_about = list(/datum/gas/oxygen, /datum/gas/water_vapor, /datum/gas/plasma, /datum/gas/carbon_dioxide, /datum/gas/nitrous_oxide, /datum/gas/nitrogen, /datum/gas/pluoxium, /datum/gas/tritium, /datum/gas/bz, /datum/gas/freon, /datum/gas/hydrogen)
```

Comment:
```suggestion
	var/list/gases_we_care_about = list(
		/datum/gas/oxygen,
		/datum/gas/water_vapor,
		/datum/gas/plasma,
		/datum/gas/carbon_dioxide,
		/datum/gas/nitrous_oxide,
		/datum/gas/nitrogen,
		/datum/gas/pluoxium,
		/datum/gas/tritium,
		/datum/gas/bz,
		/datum/gas/freon,
		/datum/gas/hydrogen,
	)
```
maybe format these lists like this for readability

Reply:
yes including the last comma

Reply:
I thought about it, but I was worried about bloating the file and obscuring the purpose of the list. I'll give it a shot

Reply:
@LemonInTheDark you did it wrong.

Reply:
I see, gottcha

Reply:
This do it?

Reply:
This work?

Reply:
yes